087-external-certificate-manager.md

```diff
@@ -0,0 +1,324 @@
+4. Once new end-entity certificates are issued that trust the new Cluster CA public cert, copy over the new Kafka certificates, updating the `strimzi.io/cluster-ca-cert-generation` annotation on the Secrets.
+5. Roll the Kafka pods to use the new Kafka certificates, updating the `strimzi.io/cluster-ca-cert-generation` annotations on the pods.
+6. Then on the next reconciliation, since the Kafka pods now have correct cert and key generation, and the Kafka certificate Secrets have the correct cert generation, copy over the new operator certificate.
+7. Since the Kafka pods now have correct cert generation, and the Kafka certificate Secrets have the correct cert generation, remove the old CA public cert from the `<CLUSTER_NAME>-cluster-ca-cert` Secret.
+
+**Note:** The above steps require a change to the CaReconciler.
+Today we only check the `strimzi.io/cluster-ca-cert-generation` annotation on the Kafka pods to decide whether to update the operator certificate Secret and remove the old Cluster CA cert.
+When using cert-manager to manage certificates, Strimzi will also check the annotation on the Kafka certificate Secrets before making either of these changes.
+
+> ![Replacing the cluster CA private key](./images/087-new-cluster-ca-key-replacement.png)
+>
+> Fig 3: Proposed workflow when the user provides a new cluster CA public cert that has been signed by a new private key
+
+### User Operator/Clients CA
+
+When a new Kafka cluster is created Strimzi will copy the CA public cert from the Secret identified in `clientsCa.publicCert` to the `<CLUSTER_NAME>-clients-ca-cert` Secret.
+Strimzi will add two annotations to the `<CLUSTER_NAME>-cluster-ca-cert` Secret:
+* `strimzi.io/ca-cert-hash` with the value being a hash of the certificate.
+* `strimzi.io/ca-cert-generation` initially set to 0.
+
+#### Issuing KafkaUser certificates
+
+The User operator will be updated to have four environment variables to configure the CA issuer:
+- `STRIMZI_CA_TYPE` with either the value `strimzi.io` or `cert-manager.io`
+- `STRIMZI_CM_ISSUER_NAME`
+- `STRIMZI_CM_ISSUER_KIND`
+- `STRIMZI_CM_ISSUER_GROUP`
+
+These will be set by the Cluster operator when the User operator is deployed as part of the Entity operator and when `spec.clientsCa.type` is set to `cert-manager.io`.
+
+When a `KafkaUser` with `spec.authentication.type` set to `tls` is created Strimzi will create a `Certificate` custom resource.
+Strimzi will specify the required CN/SANs in the `Certificate` resource for the user certificate.
+Strimzi will specify the `Secret` for the certificate to be stored to match the name of the `KafkaUser` resource.
+Strimzi will request the certificate in both PEM and PKCS12 format.
+Strimzi will wait for the usual operation timeout during the reconciliation loop for the `Certificate` status to indicate that the certificate has been issued before continuing.
```

Comment:
All the above "Strimzi" is the User Operator, can we mention it instead of Strimzi?